Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/InstanceSynchronizationConfig.swift

```diff
@@ -129,4 +76,25 @@ internal class InstanceSynchronization: Sequence {
             return nil
         }
     }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
```

Comment:
[major race condition, but might be worth opening a separate ticket] I think you need to take a `instanceLock.read` lock over this entire method. It would be pretty major race condition if this document config could change while it was being persisted, ending up with a partially corrupt persisted config. This is going to be tricky without reentrant lock semantics so it might be worth opening a ticket to have our locks have working reentrant semantics